src/errors.rs

```diff
@@ -1,62 +1,41 @@
+use std::io::Error as IoError;
+
 use protobuf::Error as ProtobufError;
 use rustls::pki_types::InvalidDnsNameError;
 use serde_json::error::Error as SerializationError;
-use std::{
-    error::Error as StdError,
-    fmt::{Display, Formatter, Result},
-    io::Error as IoError,
-};
+use thiserror::Error;
 
 /// Consolidates possible error types that can occur in the lib.
-#[derive(Debug)]
+#[derive(Debug, Error)]
 pub enum Error {
     /// This variant is used when error occurs in the lib logic.
+    #[error("an internal error occurred, {}", _0)]
```

Comment:
__question:__ wouldn't this be a bit more readable?
```suggestion
    #[error("an internal error occurred, {0}")]
```

Reply:
Oh wow, I didn't notice that was also a valid option within `thiserror` crate.
I'll update it with the proposed change.